sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/availabilitystrategy/AvailabilityStrategy.java

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.availabilitystrategy;
+
+import java.util.List;
+
+/**
+ * The type Availability strategy.
+ */
+public abstract class AvailabilityStrategy {
+    /**
+     * The Effective retry regions.
+     */
+    List<String> effectiveRetryRegions;
+    /**
+     * The Number of regions to try.
+     */
+    int numberOfRegionsToTry;
+
+    public abstract List<String> getEffectiveRetryRegions(List<String> preferredRegions, List<String> excludeRegions);
```

Comment:
Should be internal API -

Reply:
Probbaly also internal? NumberOfRegions could be passed in as paarameter as well (also easier assuming it might become mutable eventually).

Reply:
Thinking about it - I think i am fine leaving it as public API as well - but number of regions should also be passed in to allow for making it mutable eventually

Reply:
Makes sense. Will change it

Reply:
On a second thought, number of regions currently has a setter on availability strategy. So users can still set it when they extend the class